task_manager/task_manager.hh

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "replica/database.hh"
```

Comment:
This is a very heavy include. Is there a way to avoid it, get away with just forward declarations?